### PR TITLE
patch: improve chart logic and make chart responsive to filters #1361…

### PR DESCRIPTION
**Pull request by tinashechiraya**
…, #1373

[Screencast from 22-10-2023 08:19:35.webm](https://github.com/kartoza/sawps/assets/70011086/7ab88a34-2eef-49fd-9e97-530f8b65b746)

Description:
I have updated the code on the backend to preventing combining properties that are different in categories
I have updated the frontend to handle the logic
I have also made the chart responsive to the filters

### CODE REVIEW

File: django_project/frontend/utils/metrics.py [hunk 1]
```diff
@@ -98,12 +98,13 @@ def calculate_population_categories(
         owned_species__property__id__in=property_ids,
         owned_species__taxon__scientific_name=species_name
     ).values(
-        'year'
+        'year',
+        'owned_species__property__id'  # Include property_id in grouping
```
Comment (dimasciput) [inline django_project/frontend/utils/metrics.py:102]: @tinashechiraya why do we need property_id here?

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 9 min 40 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 9 min 25 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** FAILURE
**Build duration:** 58 sec

**Pull request closed without merging**